libraries/AP_Logger/AP_Logger_Block.cpp

```diff
@@ -688,21 +831,38 @@ void AP_Logger_Block::io_timer(void)
             SectorErase(next_sector / sectors_in_64k);
             next_sector += sectors_in_64k;
         }
-        gcs().send_text(MAV_SEVERITY_WARNING, "Log recovery complete, erased %d blocks", unsigned(blocks_erased));
+        status_msg = StatusMessage::RECOVERY_COMPLETE;
         df_EraseFrom = 0;
     }
 
-    if (!CardInserted() || !log_write_started) {
+    if (!CardInserted() || new_log_pending) {
         return;
     }
 
-    while (writebuf.available() >= df_PageSize - sizeof(struct PageHeader)) {
+    // write at most one page
+    if (writebuf.available() >= df_PageSize - sizeof(struct PageHeader)) {
+        WITH_SEMAPHORE(sem);
+        // if we have filled the card, stop writing
+        if (df_Write_FilePage > df_NumPages) {
+            status_msg = StatusMessage::LOGGING_STOPPED;
+            return;
+        }
         struct PageHeader ph;
-        ph.FileNumber = df_FileNumber;
-        ph.FilePage = df_FilePage;
+        ph.FileNumber = df_Write_FileNumber;
+        ph.FilePage = df_Write_FilePage;
+#if BLOCK_LOG_VALIDATE
+        ph.crc = DF_LOGGING_FORMAT;
+#endif
         memcpy(buffer, &ph, sizeof(ph));
         writebuf.read(&buffer[sizeof(ph)], df_PageSize - sizeof(ph));
         FinishWrite();
-        df_FilePage++;
+        df_Write_FilePage++;
     }
 }
+
+/*
+| Type | Size | 10Hz(bs) | 25Hz(bs) | 400Hz(Kbs) |
```

Comment:
Table format issues

Reply:
.... also, a comment as to what this is for would be good.

Reply:
I'll probably take it out - it was just notes for myself

Reply:
Removed